Start style-guide for LoopBack 4

bajtos opened the pull request:
- Copy style-guide.md to style-guide-es6.md
 - Update references, "see also" sidebars, etc. to mention LoopBack 4  and include links to ES6 guide for LoopBack 3.x

~~There are no changes of the actual content of the style guides.~~

bajtos committed:
Start style-guide for LoopBack 4

 - Copy style-guide.md to style-guide-es6.md
 - Update references, "see also" sidebars, etc. to mention LoopBack 4
   and include links to ES6 guide for LoopBack 3.x

There are no changes of the actual content of the style guides.

b-admike commented on pages/en/contrib/style-guide-es6.md:
```diff
@@ -0,0 +1,764 @@
+### Email examples
+
+- All test-related email examples should be of the format `email@example.com`.
+- The `example.com` domain was created to be used for examples in documents, and could be used without prior coordination or asking for permission.
```
This is a good rule. Did not know about it 😅

b-admike commented on pages/en/contrib/style-guide-es6.md:
````diff
@@ -0,0 +1,764 @@
+```
+
+### Hooks
+
````
It'd be great if you can add an intro to this subsection as you have done with the others.

bajtos replied:
Will do.

b-admike commented on pages/en/contrib/style-guide-es6.md:
````diff
@@ -0,0 +1,764 @@
+});
+```
+
+**Bad**:
````
This `bad` code style snippet is the same as its `good` counterpart above.

shimks replied:
Unless it's been changed, the differences in the codes are `describe` and `context`, same for the one down below

bajtos replied:
I'll add a comment to both code examples.

b-admike commented on pages/en/contrib/style-guide-es6.md:
````diff
@@ -0,0 +1,764 @@
+```
+
+```js
+context('Model', () => {
````
was this meant to be here or another duplicate?

bajtos replied:
I think this was meant to be here, it shows that `context` should not be used instead of `describe`.

dhmlau commented on pages/en/contrib/style-guide-es6.md:
````diff
@@ -0,0 +1,764 @@
+**Good**:
+
+```
+TBD
````
I guess you meant to put some content in here?

bajtos replied:
I'll remove this section in favour of "Still acceptable" below.

dhmlau commented on pages/en/contrib/style-guide-es6.md:
````diff
@@ -0,0 +1,764 @@
+```js
+describe('strong-error-handler', () => {
+  before(setupHttpServerAndClient);
+  beforeEach(resetRequestHandler)
````
not sure if it matters.. is it supposed to have a trailing `;`

bajtos replied:
Good catch!

dhmlau commented on pages/en/contrib/style-guide-es6.md:
````diff
@@ -0,0 +1,764 @@
+```js
+describe('strong-error-handler', () => {
+  before(setupHttpServerAndClient);
+  beforeEach(resetRequestHandler)
````
same here.. not sure if it matters.. is it supposed to have a trailing `;`?

bajtos commented:
@b-admike @dhmlau Thank you for reviewing the content. The issues you pointed out were old ones already present in our current style guide, but since we are touching this file, I agree it's a good idea to fix them. 

I pushed a new commit df68bc8 that addresses your comments in both `style-guide.md` and `style-guide-es6.md`.

Please note that my intent with this pull request is to make sure we have a fork of the current style guide that will remain mostly frozen for LoopBack 3.x, so that we can start updating the "current" style guide to match our new best practices in loopback-next. Those updates are out of scope of this pull request though! 

LGTY now?
